src/internal/packager/helm/post-render.go

```diff
@@ -167,22 +178,16 @@ func (r *renderer) editHelmResources(resources []releaseutil.Manifest, finalMani
 
 		switch rawData.GetKind() {
 		case "Namespace":
-			var namespace corev1.Namespace
+			namespace := new(corev1.Namespace)
 			// parse the namespace resource so it can be applied out-of-band by zarf instead of helm to avoid helm ns shenanigans
-			if err := runtime.DefaultUnstructuredConverter.FromUnstructured(rawData.UnstructuredContent(), &namespace); err != nil {
+			if err := runtime.DefaultUnstructuredConverter.FromUnstructured(rawData.UnstructuredContent(), namespace); err != nil {
 				message.WarnErrf(err, "could not parse namespace %s", rawData.GetName())
 			} else {
 				message.Debugf("Matched helm namespace %s for zarf annotation", namespace.Name)
-				if namespace.Labels == nil {
-					// Ensure label map exists to avoid nil panic
-					namespace.Labels = make(map[string]string)
-				}
-				// Now track this namespace by zarf
-				namespace.Labels[config.ZarfManagedByLabel] = "zarf"
-				namespace.Labels["zarf-helm-release"] = r.chart.ReleaseName
```

Comment:
Just noticed this label while I was looking through the code. It comes into play when there are manifests or helm charts that create a namespace separate from the declared namespace. Maybe it helps when people have helm charts that release to multiple namespaces (though that's an antipattern)? But it only goes into extra namespaces that aren't declared in the Zarf chart. The label also isn't stripped on Zarf destroy. 

If we keep it I think we should start adding it to Zarf declared namespaces as well, which I'd be fine with.